### PR TITLE
SoapClientFactory

### PR DESCRIPTION
**Pull request by hranicka**
The code is not testable and mainly I cannot set custom SOAP config.

For example I have problems with SSL certificates.
But it's impossible set custom options and configure certificates options/paths.

So I've made only necessary changes and allowed set custom SoapClient factory.

See tests.

### CODE REVIEW

File: src/GopayConfig.php [hunk 1]
```diff
@@ -14,7 +14,13 @@ class GopayConfig
 	/**
 	 * Parametr specifikujici, pracuje-li se na testovacim ci provoznim prostredi
 	 */
-	static $version = self::TEST;
+	public static $version = self::TEST;
+
+	/**
+	 * @var callable
+	 */
+	public static $soapClientFactory = '\Markette\Gopay\Api\SoapClientFactory::create';
```
Comment (hranicka) [inline src/GopayConfig.php]: Almost everything in API is static.

So I've added static property with factory.

The default one simply creates `new \SoapClient($wsdl, array())` which was created on many places before.

**Comment by f3l1x** [comment]
It's great. Could you please revert reformating? Just add a new feature. Thx.

**Comment by hranicka** [comment]
@f3l1x Ok, sorry. I've deleted the "typo" commit.

It should be clean PR now :)

**Comment by hranicka** [comment]
It's not a BC-break. I think that these changes were needed:

* tests (mocking of SoapClient)
* custom config of SoapClient for every application (eg. SSL certificates)

**Comment by f3l1x** [comment]
I dont understand.

**Comment by hranicka** [comment]
@f3l1x sorry, I' ve fixed the comment now.

**Pull request merged**